x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -299,22 +301,44 @@ public void createApiKey(
         if (authentication == null) {
             listener.onFailure(new IllegalArgumentException("authentication must be provided"));
         } else {
-            createApiKeyAndIndexIt(authentication, request, userRoleDescriptors, listener);
+            final TransportVersion version = getMinNodeTransportVersion();
+            if (version.before(Authentication.VERSION_API_KEYS_WITH_REMOTE_INDICES) && hasRemoteIndices(request.getRoleDescriptors())) {
+                // Creating API keys with roles which define remote indices privileges is not allowed in a mixed cluster.
+                listener.onFailure(
+                    new IllegalStateException(
```

Comment:
I think this should be an `IllegalArgumentException` (maps to `400`) because it's something that end-user can fix by changing the request.

Reply:
Good catch! I'll change this in the [follow up PR](https://github.com/elastic/elasticsearch/pull/93808) that I'm working on.